### DISCUSSION TITLE
callback_method query param missing from new documentation

### DISCUSSION THREAD

**Discussion by navin-statisfy** [deepgram/community#513] [General help] [answered]
### Which Deepgram product are you using?

Deepgram SDKs

### Details

I am trying to use [s3 presigned urls](https://developers.deepgram.com/docs/using-aws-s3-presigned-urls-with-the-deepgram-api#uploading-transcripts-directly-to-s3), but don't see any way to specify callback_method in the [api reference](https://developers.deepgram.com/reference/listen)/python sdk

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jkroll-deepgram** [comment]
Hi @navin-statisfy, please use the `callback` field (from the [API reference](https://developers.deepgram.com/reference/listen) you linked), or the `transcribe_file_callback()` method in the [Python v3 SDK](https://github.com/deepgram/deepgram-python-sdk/blob/5a29caedd097bdffe7e8d03536ff119ffd29bfca/deepgram/clients/prerecorded/v1/client.py#L215), with parameter `callback`.

I agree we could make the Python SDK callbacks clearer, I will facilitate some improved documentation with our new Python SDK version.

**Reply by navin-statisfy** [reply, marked as answer]
@jkroll-deepgram : Please see the attached screenshot from [S3 presigned documentation](https://developers.deepgram.com/docs/using-aws-s3-presigned-urls-with-the-deepgram-api#uploading-transcripts-directly-to-s3) mentioning that default callback method is POST and we need to specify it as PUT for this to work.
![Screenshot 2024-01-03 at 10 46 04 AM](https://github.com/deepgram/community/assets/151481745/285775f7-2b05-4de4-8bba-77df18b6a034)

Here are the two request IDs that I tried with signed url as PUT & POST and it didn't work.

8d836f72-44c9-4691-aab7-a1ea91d86e35 (POST)
4f26a2d5-a8cc-4b08-88c5-a670973bec73 (PUT)

Note: I am using GCS signed urls instead of S3, I don't think that should make a difference.

**Reply by jjmaldonis** [reply]
@navin-statisfy, you can add the `callback_method=put` [query parameter](https://developers.deepgram.com/docs/callback#pre-recorded-audio) to your URL to ask Deepgram to send the callback request as a PUT rather than a POST.

**Reply by navin-statisfy** [reply]
Got it. So I'll need to use the rest api directly for this & can't use the python SDK since the option is not exposed yet.

**Reply by jjmaldonis** [reply]
I just created a PR to add the query param to the python SDK, see https://github.com/deepgram/deepgram-python-sdk/pull/251. If you "Watch" this PR it should get merged soon and you can update (you may need to use pip to install directly from GitHub because we probably won't do a release for a few days).

**Reply by mjabl** [reply]
BTW, the C# SDK seems to hate the callback_method param too.

**Comment by jpvajda** [comment]
@mjabl  👋  Could you elaborate on what you mean about our .NET SDK and the callback_method? 

>  the C# SDK seems to hate the callback_method param too.

I was just curious what you meant so we can try to better understand.

**Reply by mjabl** [reply]
@jpvajda I mean you can't set callback_method on PrerecordedTranscriptionOptions class. Which effectively renders the C# sdk useless for me as I really need to receive PUT callbacks and not the default POST callbacks.

**Comment by jpvajda** [comment]
@mjabl  Thank you for clarifying. That is a feature we've not yet introduced into our .NET SDK, but intend to in the future. You can follow this issue for progress: https://github.com/deepgram/deepgram-dotnet-sdk/issues/232
